### DISCUSSION TITLE
Can clippy take into account --ignore-rust-version?

### DISCUSSION THREAD

**Discussion by Pat-Lafon** [rust-lang/rust-clippy#10709] [General]
With the ability of clippy to check `rust-version` from the `Cargo.toml` https://github.com/rust-lang/rust-clippy/issues/8746, I was wondering about how this interacts with `--ignore-rust-version`. It looks like this flag is just passed along and ignored on the clippy side?

If that's the case I was wondering if it would make sense for clippy to have some handling for this flag to turn off the version check?

This came up in https://github.com/lalrpop/lalrpop/pull/763#issuecomment-1519121505 where for lalrpop, the suggestion is to set the msrv as somewhere <= stable -2 and then increase it within that constraint as new features are added. Here it would be nice if clippy could suggest lints that use these new features, even if they are not compatible with the current version.

**Comment by flip1995** [comment]
IIUC the `--ignore-rust-version` flag in `cargo` is only for `cargo install`. Clippy mostly only interacts with `cargo check`, so as long as `cargo check` doesn't support this, we'll probably won't add support for it to Clippy. 

This is a cargo feature and Clippy should not re-implement cargo features.

**Reply by Pat-Lafon** [reply]
I believe `cargo check` does support `--ignore-rust-version`? https://doc.rust-lang.org/cargo/commands/cargo-check.html#compilation-options

**Reply by flip1995** [reply]
Awesome, then this should be supported out of the box. Clippy reads the rust-version specified in the `Cargo.toml` file from the [`CARGO_PKG_RUST_VERSION`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates) env var set by cargo. I think, the issue why it is not supported is, that `cargo` sets this env var (and the metadata), even though this flag is passed. So I don't think this needs any changes on the Clippy side, but rather on the `cargo` side.
